File: csharp/lib/Property/BaselineLengthProperty.cs
/*
 * SPDX-FileCopyrightText: 2017-2022 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

using MessagePack;

namespace SensCord
{
    /// <summary>
    /// Structure for handling baseline length between cameras.
    /// </summary>
    [MessagePackObject]
    public class BaselineLengthProperty : IBaseProperty
    {
        /// <summary>
        /// Property key.
        /// </summary>
        public static readonly string ConstKey = "baseline_length_property";

        /// <summary>
        /// Property key.
        /// </summary>
        [IgnoreMember]
        public string Key { get; } = ConstKey;

        /// <summary>
        /// Baseline length in millimeters.
        /// </summary>
        [Key("length_mm")]
        public float Length { get; set; }

        /// <summary>
        /// To string.
        /// </summary>
        /// <returns>A string that represents the current object.</returns>
        public override string ToString()
        {
            return $"{this.Key}(Length={this.Length})";
        }
    }
}

File: java/senscord/src/senscord_properties/BaselineLengthProperty.java
/*
 * SPDX-FileCopyrightText: 2019 Sony Semiconductor Solutions Corporation
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package senscord_properties;

import com.sun.jna.Structure;
import java.util.Arrays;
import java.util.List;

/**
 * Structure for handling baseline length between cameras.
 * 
 * @see "senscord::BaselineLengthProperty"
 */
public class BaselineLengthProperty extends Structure {
    public float length_mm;

    public static class ByReference extends BaselineLengthProperty
            implements Structure.ByReference {
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("length_mm");
    }

    /**
     * Get baseline length. Unit is millimeters.
     * 
     * @return baseline length
     */
    public float GetLength() {
        return length_mm;
    }

    /**
     * Set baseline length. Unit is millimeters.
     * 
     * @param length_mm baseline length
     */
    public void SetLength(float length_mm) {
        this.length_mm = length_mm;
    }
}
